include/rocksdb/status.h

```diff
@@ -73,11 +73,24 @@ class Status {
     kDeadlock = 5,
     kStaleFile = 6,
     kMemoryLimit = 7,
+    kSpaceLimit = 8,
     kMaxSubCode
   };
 
   SubCode subcode() const { return subcode_; }
 
+  enum Severity {
+    kNoError = 0,
+    kSoftError = 1,
+    kHardError = 2,
+    kFatalError = 3,
+    kUnrecoverableError = 4,
+    kMaxSeverity
+  };
```

Comment:
How many bytes does it take by default? If more than 1 maybe it's a good idea to explicitly claim it to be a byte, just as what we do with `PerfLevel`.

Reply:
Default enum is 4 bytes I think? I don't mind setting it to unsigned char. Code and SubCode are default size, though, so it might look a little inconsistent.

Reply:
I mean change all of them.

Reply:
What do you hope to optimize? The Status copy constructor explicitly copies each field, so I don't think we'll gain anything in terms of CPU. It is mostly used on stack, so it won't change memory usage either.